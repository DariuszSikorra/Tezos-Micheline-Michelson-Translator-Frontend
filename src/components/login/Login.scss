.login {
    .login-content {
        max-width: 500px;
        margin-left: auto;
        margin-right: auto;
        justify-content: center;
    }
}
