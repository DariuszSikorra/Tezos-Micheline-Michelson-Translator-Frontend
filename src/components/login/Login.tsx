import React from "react";
import { Formik } from "formik";
import * as Yup from "yup";
import { Redirect } from "react-router-dom";
import { useMappedState, useDispatch } from "redux-react-hook";
import * as authTypes from "../../store/authentication/types";
import { IState } from "../../store/global/types";
import FormInput from "../shared/input/FormInput";
import FormButton from "../shared/formButton/FormButton";
import "./Login.scss";

const mapState = (state: IState) => ({
  auth: state.auth,
});

const validationSchema = Yup.object().shape({
  login: Yup.string().required("Login is needed"),
  password: Yup.string().required("Password is needed"),
});

const Login = () => {
  const { auth } = useMappedState(mapState);
  const dispatch = useDispatch();

  const submitForm = (values: authTypes.authCredentials) => {
    dispatch({
      type: authTypes.AUTHENTICATION_LOGIN,
      payload: values,
    });
  };
  const redirectToPanel = () => auth.isLogged && <Redirect to="/admin/library" />;

  return (
    <div className="login">
      <div className="login-content">
        {redirectToPanel()}
        <Formik
          initialValues={{
            username: "",
            password: "",
          }}
          validationSchema={validationSchema}
          onSubmit={submitForm}
        >
          {({ values, errors, touched, handleChange, handleBlur, handleSubmit, isSubmitting }) => (
            <form onSubmit={handleSubmit} noValidate>
              <FormInput
                label="Login"
                type="text"
                name="username"
                onChange={handleChange}
                onBlur={handleBlur}
                value={values.username}
                errors={errors.username}
                touched={touched.username}
              />
              <FormInput
                label="Password"
                type="password"
                name="password"
                onChange={handleChange}
                onBlur={handleBlur}
                value={values.password}
                errors={errors.password}
                touched={touched.password}
              />
              <FormButton
                label="Log in"
                type="submit"
                stylingType="submit"
                onClick={() => submitForm(values)}
                disabled={isSubmitting || !!errors}
              />
            </form>
          )}
        </Formik>
      </div>
    </div>
  );
};

export default Login;
