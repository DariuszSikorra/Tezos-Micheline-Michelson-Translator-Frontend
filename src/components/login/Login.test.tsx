import React from "react";
import { mount } from "enzyme/build";
import { StoreContext } from "redux-react-hook";
import Login from './Login';

import configureStore from "../../store/index";
import rootSaga from "../../store/saga";

const store = configureStore();
store.runSaga(rootSaga);

describe("Login", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(
            <StoreContext.Provider value={store}>
                <Login />
            </StoreContext.Provider>,
        );
    });

    it("matches snapshot", () => {
        expect(wrapper.debug()).toMatchSnapshot();
    })
})