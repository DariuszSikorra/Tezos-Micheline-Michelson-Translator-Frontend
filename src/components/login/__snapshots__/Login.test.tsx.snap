// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches snapshot 1`] = `
"<Login>
  <div className=\\"login\\">
    <div className=\\"login-content\\">
      <Formik initialValues={{...}} validationSchema={{...}} onSubmit={[Function: submitForm]} validateOnChange={true} validateOnBlur={true} isInitialValid={false} enableReinitialize={false}>
        <form onSubmit={[Function]} noValidate={true}>
          <FormInput label=\\"Login\\" type=\\"text\\" name=\\"login\\" onChange={[Function]} onBlur={[Function]} value=\\"\\" errors={[undefined]} touched={[undefined]}>
            <div className=\\"form-input undefined\\">
              <p className=\\"form-input_label\\">
                Login
              </p>
              <input name=\\"login\\" type=\\"text\\" onChange={[Function]} onBlur={[Function]} placeholder=\\"Login\\" value=\\"\\" test-id=\\"input-test\\" className=\\"form-input_input\\" />
              <div className=\\"form-input_validation-marker undefined\\" />
              <div className=\\"form-input_error\\" />
            </div>
          </FormInput>
          <FormInput label=\\"Password\\" type=\\"password\\" name=\\"password\\" onChange={[Function]} onBlur={[Function]} value=\\"\\" errors={[undefined]} touched={[undefined]}>
            <div className=\\"form-input undefined\\">
              <p className=\\"form-input_label\\">
                Password
              </p>
              <input name=\\"password\\" type=\\"password\\" onChange={[Function]} onBlur={[Function]} placeholder=\\"Password\\" value=\\"\\" test-id=\\"input-test\\" className=\\"form-input_input\\" />
              <div className=\\"form-input_validation-marker undefined\\" />
              <div className=\\"form-input_error\\" />
            </div>
          </FormInput>
          <FormButton type=\\"submit\\" label=\\"Log in\\" onClick={[Function: onClick]}>
            <button className=\\"form-button form-button_submit undefined\\" data-testid=\\"form-button\\" onClick={[Function: onClick]} disabled={[undefined]}>
              Log in
            </button>
          </FormButton>
        </form>
      </Formik>
    </div>
  </div>
</Login>"
`;
